### ISSUE TITLE
Tracing throws NullRefException when using null parameter

### ISSUE THREAD

**Issue by hutterm** [jonwagner/EventSourceProxy#43]
Here is a minimal test case - it throws on the last line:

```
    public class Test
    {
        public int Int;
    }

    public interface IInterface
    {
        void TraceTest(Test t);
    }

    [TestMethod]
    public void TraceTest()
    {
        TraceParameterProvider.Default.ForAnything().With<Test>().Trace(test => test.Int).EndWith();
        var iinterface = EventSourceImplementer.GetEventSourceAs<IInterface>();
        var _listener = new TestEventListener();
        _listener.EnableEvents((EventSource)iinterface, EventLevel.LogAlways);
        iinterface.TraceTest(null);
    }
```

I haven't found a clever way to get around that easily. Can you handle this special case internally?

**Comment by mjeanrichard** [comment]
I think you could use 
```TraceParameterProvider.Default.ForAnything().With<Test>().Trace(test => test == null ? 0 : test.Int).EndWith()```
That would prevent the NullRef.

**Comment by jonwagner** [comment]
This is fixed in 4.0.0 running against net46/netstandard2.0
